src/target/riscv_debug.c

```diff
@@ -827,24 +827,28 @@ static target_halt_reason_e riscv_halt_poll(target_s *const target, target_addr_
 static void riscv_reset(target_s *const target)
 {
 	riscv_hart_s *const hart = riscv_hart_struct(target);
+	bool has_reset = false;
 	/* If the target does not have the nRST pin inhibited, use that to initiate reset */
 	if (!(target->target_options & RV_TOPT_INHIBIT_NRST)) {
 		platform_nrst_set_val(true);
-		riscv_dm_poll_state(hart->dbg_module, RV_DM_STAT_ALL_RESET);
+		has_reset = riscv_dm_poll_state(hart->dbg_module, RV_DM_STAT_ALL_RESET);
 		platform_nrst_set_val(false);
 		/* In theory we're done at this point and no debug state was perturbed */
-	} else {
-		/*
-		 * Otherwise, if nRST is not usable, use instead reset via dmcontrol. In this case,
-		 * when reset is requested, use the ndmreset bit to perform a system reset
-		 */
-		riscv_dm_write(hart->dbg_module, RV_DM_CONTROL, hart->hartsel | RV_DM_CTRL_SYSTEM_RESET);
+	}
+
+	/*
+		* Otherwise, if nRST is not usable (or if it failed), use instead reset via dmcontrol. In this case,
+		* when reset is requested, use the ndmreset bit to perform a system reset
+		*/
```

Comment:
This comment needs slightly reformatting - your editor removed the spaces that indent the *'s on the continuation lines instead of the tabs, hence the odd appearance.

Reply:
indeed,  i tried to fix that, it looks ok here now (?)

Reply:
The whitespace has been entirely converted to spaces - the indentation should be tabs (1x) followed by a single space to align the *'s in the comment block.

Reply:
![image](https://github.com/blackmagic-debug/blackmagic/assets/691140/57961921-6c57-4a1e-9a65-0c2505702bf6)
vs
![image](https://github.com/blackmagic-debug/blackmagic/assets/691140/ed3b3d45-0a5a-4f0e-8f75-f4690407f478)